examples/vision/oxford_pets_image_segmentation.py

```diff
@@ -116,129 +221,281 @@ def get_model(img_size, num_classes):
-        x = layers.Conv2DTranspose(filters, 3, padding="same")(x)
-        x = layers.BatchNormalization()(x)
+        x = keras.layers.Activation("relu")(x)
+        x = keras.layers.Conv2DTranspose(filters, 3, padding="same")(x)
+        x = keras.layers.BatchNormalization()(x)
 
-        x = layers.Activation("relu")(x)
-        x = layers.Conv2DTranspose(filters, 3, padding="same")(x)
-        x = layers.BatchNormalization()(x)
+        x = keras.layers.Activation("relu")(x)
+        x = keras.layers.Conv2DTranspose(filters, 3, padding="same")(x)
+        x = keras.layers.BatchNormalization()(x)
 
-        x = layers.UpSampling2D(2)(x)
+        x = keras.layers.UpSampling2D(2)(x)
 
         # Project residual
-        residual = layers.UpSampling2D(2)(previous_block_activation)
-        residual = layers.Conv2D(filters, 1, padding="same")(residual)
-        x = layers.add([x, residual])  # Add back residual
+        residual = keras.layers.UpSampling2D(2)(previous_block_activation)
+        residual = keras.layers.Conv2D(filters, 1, padding="same")(residual)
+        x = keras.layers.add([x, residual])  # Add back residual
         previous_block_activation = x  # Set aside next residual
 
     # Add a per-pixel classification layer
-    outputs = layers.Conv2D(num_classes, 3, activation="softmax", padding="same")(x)
+    outputs = keras.layers.Conv2D(num_classes, 3, activation="softmax", padding="same")(
+        x
+    )
 
     # Define the model
     model = keras.Model(inputs, outputs)
     return model
 
 
-# Free up RAM in case the model definition cells were run multiple times
-keras.backend.clear_session()
-
-# Build model
-model = get_model(img_size, num_classes)
-model.summary()
+# Taking a batch of test inputs to measure model's progress.
+test_images, test_masks = next(iter(resized_val_ds))
 
 """
-## Set aside a validation split
+Let us define a custom learning rate scheduler for the U-Net model we created, which uses
+a warmup phase followed by cosine decay, in order to help improve the training of the
+model later.
+
+We start with a small learning rate (warmup phase), gradually increase it to avoid large
+initial updates, and then slowly decrease it following a cosine schedule after the warmup
+phase, allowing the learning rate to explore different regions of the weight space and
+aiding in achieving a good solution.
 """
 
-import random
 
-# Split our img paths into a training and a validation set
-val_samples = 1000
-random.Random(1337).shuffle(input_img_paths)
-random.Random(1337).shuffle(target_img_paths)
-train_input_img_paths = input_img_paths[:-val_samples]
-train_target_img_paths = target_img_paths[:-val_samples]
-val_input_img_paths = input_img_paths[-val_samples:]
-val_target_img_paths = target_img_paths[-val_samples:]
-
-# Instantiate data Sequences for each split
-train_gen = OxfordPets(
-    batch_size, img_size, train_input_img_paths, train_target_img_paths
+class WarmUpCosine(
```

Comment:
Did this help ? I still see a large "hump" in the validation loss.
If this LR schedule does not produce a marked improvement, I'd suggest not to use it.
The LR Schedule definition below is 80+ lines of code which is a lot and takes the focus away from U-NET and Keras-CV.
If the LR schedule really is useful and you want to keep it, I'd recommend going for a simpler tf.keras.optimizers.schedules.PiecewiseConstantDecay with 4 or 5 values for the LR. It will be 2 lines of code.

Reply:
We have decided to remove the schedule and also reove the plots from the tutorial.